azkaban-common/src/main/java/azkaban/executor/Statistics.java

```diff
@@ -0,0 +1,186 @@
+      this.remainingMemoryPercent = value;
+    }
+
+    public long getRemainingMemory(){
+      return this.remainingMemory;
+    }
+
+    public void setRemainingMemory(long value){
+      this.remainingMemory = value;
+    }
+
+    public int getRemainingFlowCapacity(){
+      return this.remainingFlowCapacity;
+    }
+
+    public void setRemainingFlowCapacity(int value){
+      this.remainingFlowCapacity = value;
+    }
+
+    public Date getLastDispatchedTime(){
+      return this.lastDispatchedTime;
+    }
+
+    public void setLastDispatchedTime(Date value){
+      this.lastDispatchedTime = value;
+    }
+
+    public long getRemainingStorage() {
+      return this.remainingStorage;
+    }
+
+    public void setRemainingStorage(long value){
+      this.remainingStorage = value;
+    }
+
+    public int getPriority () {
+      return this.priority;
+    }
+
+    public void setPriority (int value) {
+      this.priority = value;
+    }
+
+    public int getNumberOfAssignedFlows () {
+      return this.numberOfAssignedFlows;
+    }
+
+    public void setNumberOfAssignedFlows (int value) {
+      this.numberOfAssignedFlows = value;
+    }
+
+    public Statistics(){}
+
+    public Statistics (double remainingMemoryPercent,
+        long remainingMemory,
+        int remainingFlowCapacity,
+        Date lastDispatched,
+        long remainingStorage,
+        double cpuUsage,
+        int  priority,
+        int numberOfAssignedFlows){
+      this.remainingMemory = remainingMemory;
+      this.cpuUsage = cpuUsage;
+      this.remainingFlowCapacity = remainingFlowCapacity;
+      this.priority = priority;
+      this.remainingMemoryPercent = remainingMemoryPercent;
+      this.remainingStorage = remainingStorage;
+      this.lastDispatchedTime = lastDispatched;
+      this.numberOfAssignedFlows = numberOfAssignedFlows;
+    }
+
+    @Override
+    public boolean equals(Object obj)
+    {
+        if (obj instanceof Statistics)
+        {
+          boolean result = true;
+          Statistics stat = (Statistics) obj;
+
+          result &=this.remainingMemory == stat.remainingMemory;
+          result &=this.cpuUsage == stat.cpuUsage;
+          result &=this.remainingFlowCapacity == stat.remainingFlowCapacity;
+          result &=this.priority == stat.priority;
+          result &=this.remainingMemoryPercent == stat.remainingMemoryPercent;
+          result &=this.remainingStorage == stat.remainingStorage;
+          result &=this.numberOfAssignedFlows == stat.numberOfAssignedFlows;
+          result &= null == this.lastDispatchedTime ? stat.lastDispatchedTime == null :
+                            this.lastDispatchedTime.equals(stat.lastDispatchedTime);
+          return result;
+        }
+        return false;
+    }
+
+
+    // really ugly to have it home-made here for object-binding as base on the
+    // current code base there is no any better ways to do that.
+    public static Statistics fromJsonObject(Map<String,Object> mapObj){
```

Comment:
fixed by using the Mapper directly rather than the JsonUtils,  also fixed the RestfulAPiClient to adopt the change.